code/modules/food_and_drinks/food.dm

```diff
@@ -16,13 +16,17 @@
 	resistance_flags = FLAMMABLE
 	var/foodtype = NONE
 	var/last_check_time
+	var/food_quality = 50
 
 /obj/item/reagent_containers/food/Initialize(mapload)
 	. = ..()
 	if(!mapload)
 		pixel_x = rand(-5, 5)
 		pixel_y = rand(-5, 5)
 
+/obj/item/reagent_containers/food/proc/adjust_food_quality(new_quality)
+	food_quality = clamp(new_quality,0,100)
```

Comment:
This could as well be a defined macro unless subtypes with other effects based on food quality are planned.
Also, until we finally butcher 512 compatibily (which will be soon):
```suggestion
	food_quality = CLAMP(new_quality,0,100)
```